pkg/routeagent/controllers/route/route.go

```diff
@@ -260,6 +264,40 @@ func (r *Controller) runPodWorker() {
 	}
 }
 
+func (r *Controller) overlappingSubnets(remoteSubnets []string) bool {
+	// If the remoteSubnets [*] overlap with local cluster Pod/Service CIDRs we
+	// should not update the IPTable rules on the host, as it will disrupt the
+	// functionality of the local cluster. So, lets validate that subnets do not
+	// overlap before we program any IPTable rules on the host for inter-cluster
+	// traffic.
+	// [*] Note: In a non-GlobalNet deployment, remoteSubnets will be a list of
+	// Pod/Service CIDRs, whereas in a GlobalNet deployment, it will be a list of
+	// globalCIDRs allocated to the clusters.
+	for _, serviceCidr := range r.localServiceCidr {
+		overlap, err := util.IsOverlappingCIDR(remoteSubnets, serviceCidr)
+		if err != nil {
+			// Ideally this case will never hit, as the subnets are valid CIDRs
+			klog.Warningf("unable to validate overlapping Service CIDR: %s", err)
+		}
+		if overlap {
+			klog.Errorf("Local Service CIDR %q, overlaps with remote cluster %s", serviceCidr, err)
+			return true
+		}
+	}
+
+	for _, podCidr := range r.localClusterCidr {
+		overlap, err := util.IsOverlappingCIDR(remoteSubnets, podCidr)
+		if err != nil {
+			klog.Warningf("unable to validate overlapping Pod CIDR: %s", err)
+		}
+		if overlap {
+			klog.Errorf("Local Pod CIDR %q, overlaps with remote cluster %s", podCidr, err)
```

Comment:
Will we hit this in case of globalnet?

Reply:
Yes, this will validate the third scenario mentioned in the commit message.

"In a Globalnet deployment, if localCluster Pod/Service CIDRs overlap with globalCIDR
of other clusters."